core/trino-main/src/main/java/io/trino/operator/PartitionedLookupSourceFactory.java

```diff
@@ -74,6 +74,9 @@
     @GuardedBy("lock")
     private final SettableFuture<?> destroyed = SettableFuture.create();
 
+    @GuardedBy("lock")
+    private boolean probeFinished;
```

Comment:
per https://github.com/trinodb/trino/pull/7455#discussion_r605756896 can you help me understand why do we need a new field?

Reply:
`io.trino.operator.PartitionedLookupSourceFactory#finishProbeOperator` can exist in two places. Both need to be protected from race.